pkg/controller/cache/cache_controller.go

```diff
@@ -92,10 +111,163 @@ func (r *ReconcileCache) Reconcile(request reconcile.Request) (reconcile.Result,
+			return reconcile.Result{RequeueAfter: time.Second * 10}, err
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Cluster must be well formed
+	for i, c := range ispnInstance.Status.Conditions {
+		if c.Type == "wellFormed" && c.Status == "True" {
+			break
+		}
+		if i == len(ispnInstance.Status.Conditions)-1 {
+			reqLogger.Info("Infinispan cluster " + ispnInstance.Name + " not well formed")
+			return reconcile.Result{RequeueAfter: time.Second * 10}, err
+		}
+	}
+	// Authentication is required to go on from here
+	user, pass, result, err := getCredentials(r, reqLogger, instance)
+	if err != nil {
+		return result, err
+	}
+	// List the pods for this infinispan's deployment
+	podList := &corev1.PodList{}
+	labelSelector := labels.SelectorFromSet(util.GetLabelsForInfinispan(ispnInstance.Name, ""))
+	listOps := &client.ListOptions{Namespace: ispnInstance.GetClusterName(), LabelSelector: labelSelector}
+	err = r.client.List(context.TODO(), podList, listOps)
+	if err != nil || (len(podList.Items) == 0) {
+		reqLogger.Error(err, "failed to list pods", "Infinispan.Namespace")
+		return reconcile.Result{}, err
+	}
+	cacheName := instance.Name
+	if instance.Spec.Name != "" {
+		cacheName = instance.Spec.Name
+	}
+	caches, err := cluster.CacheNames(user, pass, podList.Items[0].Name, instance.Namespace, ispnctrl.Protocol(ispnInstance))
+	if err == nil {
+		for i, cn := range caches {
+			if cn == cacheName {
+				reqLogger.Info("Cache already exists")
+				// Check if template matches?
+				break
+			}
+			if i == len(caches)-1 {
+				reqLogger.Info("Cache doesn't exist, create it")
+				podName := podList.Items[0].Name
+				xmlTemplate := instance.Spec.Template
+				if xmlTemplate == "" {
+					xmlTemplate, err = ispnctrl.GetDefaultCacheTemplateXML(podName, ispnInstance, cluster, reqLogger)
+				}
+				if err != nil {
+					reqLogger.Error(err, "Error getting default XML")
+					return reconcile.Result{}, err
+				}
+				reqLogger.Info(xmlTemplate)
+				err = cluster.CreateCacheWithAuth(user, pass, instance.Spec.Name, xmlTemplate, podName, instance.Namespace, ispnctrl.Protocol(ispnInstance))
+				if err != nil {
+					reqLogger.Error(err, "Error in creating cache")
+					return reconcile.Result{}, err
+				}
+			}
+		}
+	} else {
+		reqLogger.Error(err, "Error getting caches list")
+		return reconcile.Result{}, err
+	}
+
+	reqLogger.Info("Update CR status with connection info")
+
+	// Search the service associated to the cluster
+	serviceList := &corev1.ServiceList{}
+	labelSelector = labels.SelectorFromSet(ispnutil.GetLabelsForInfinispan(ispnInstance.Name, "infinispan-service"))
+	listOps = &client.ListOptions{Namespace: ispnInstance.Namespace, LabelSelector: labelSelector}
+	err = r.client.List(context.TODO(), serviceList, listOps)
+	if err != nil {
+		reqLogger.Error(err, "failed to select cluster service")
+		return reconcile.Result{}, err
+	}
+	if len(serviceList.Items) != 1 {
```

Comment:
@rigazilla why we need this check? If needed, we should did this check on the start?

Reply:
yeah it's probably superfluous

Reply:
removed